specifyweb/frontend/js_src/lib/components/WorkBench/WbSpreadsheet.tsx

```diff
@@ -0,0 +1,501 @@
+    closeNoDisambiguationDialog,
+  ] = useBooleanState();
+  const [disambiguationDialog, openDisambiguation, closeDisambiguation] =
+    useBooleanState();
+
+  const openDisambiguationDialog = () => {
+    if (mappings === undefined || !hotRef.current) return;
+
+    const [visualRow, visualCol] = getSelectedLast(
+      hotRef.current?.hotInstance!
+    );
+    const physicalRow = hotRef.current?.hotInstance!.toPhysicalRow(visualRow);
+    const physicalCol =
+      hotRef.current?.hotInstance!.toPhysicalColumn(visualCol);
+
+    const matches = validation?.uploadResults.ambiguousMatches[
+      physicalRow
+    ].find(({ physicalCols }) => physicalCols.includes(physicalCol));
+    if (matches === undefined) return;
+    const tableName = getTableFromMappingPath(
+      mappings.baseTable.name,
+      matches.mappingPath
+    );
+    const table = strictGetTable(tableName);
+    const resources = new table.LazyCollection({
+      filters: { id__in: matches.ids.join(',') },
+    }) as Collection<AnySchema>;
+
+    (hasTablePermission(table.name, 'read')
+      ? resources.fetch({ limit: 0 })
+      : Promise.resolve(resources)
+    ).then(({ models }) => {
+      if (models.length === 0) {
+        openNoDisambiguationDialog();
+        return;
+      }
+
+      // Re-enable this once live validation is available again:
+      /*
+       * Disable "Apply All" if validation is still in progress.
+       * This is because we don't know all matches until validation is done
+       */
+      /*
+       *Let applyAllAvailable = true;
+       *const applyAllButton = content.find('#applyAllButton');
+       *
+       *const updateIt = () => {
+       *  const newState = this.liveValidationStack.length === 0;
+       *  if (newState !== applyAllAvailable) {
+       *    applyAllAvailable = newState;
+       *    applyAllButton.disabled = !newState;
+       *    applyAllButton[newState ? 'removeAttribute' : 'setAttribute'](
+       *      'title',
+       *      wbText.applyAllUnavailable()
+       *    );
+       *  }
+       *};
+       *
+       *const interval = globalThis.setInterval(updateIt, 100);
+       * // onClose: globalThis.clearInterval(interval);
+       */
+
+      setDisambiguationMatches(matches);
+      setResource(resources);
+      setPhysicalRow(physicalRow);
+      openDisambiguation();
+    });
+  };
+
+  const contextMenuConfig = {
+    items: ensure<
+      IR<
+        | Handsontable.plugins.ContextMenu.MenuItemConfig
+        | Handsontable.plugins.ContextMenu.PredefinedMenuItemKey
+      >
+    >()(
+      isUploaded
+        ? ({
+            // Display uploaded record
+            upload_results: {
+              disableSelection: true,
+              isCommand: false,
+              renderer: (_hot, wrapper) => {
+                const { endRow: visualRow, endCol: visualCol } =
+                  getSelectedRegions(_hot).at(-1) ?? {};
+                const physicalRow = _hot.toPhysicalRow(visualRow ?? 0);
```

Comment:
function arguments starting with `_` mean they are unused (not specific to our codebase - it's a common practice in JS in general)

unused arguments are present if for example the function cares about some argument after the unused one (like `table` in this case), or if unused arguments are defined just for documentation purposes

since the variable is no longer unused, you should rename `_hot` to `hot`